### PR TITLE
Update directory-structure.md

### LINKED ISSUE #21344: For self-contained deployments (at least on ASP.NET Core 5.0) the `*.exe` extension is only generated for Windows.
For self-contained deployments (at least on ASP.NET Core 5.0) the `*.exe` extension is only generated for Windows. For Linux (and presumably macOS) no extension is generated. This behaviour is consistent with that for framework-dependent deployments so I just copied the relevant text. I am not sure if the behaviour applies to versions <= aspnetcore-3.0 so I haven't touched the text there.

It may also be useful to add that if you choose the `Produce single file` option when publishing, none of the `*.dll` files remain in the _publish_ directory (I love this feature!).

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 566f8ed0-217e-5125-b0f7-6d93c3a145c1
* Version Independent ID: 9fa94395-eff8-3b12-c2d5-c1c6d4142378
* Content: [ASP.NET Core directory structure](https://docs.microsoft.com/en-us/aspnet/core/host-and-deploy/directory-structure?view=aspnetcore-5.0)
* Content Source: [aspnetcore/host-and-deploy/directory-structure.md](https://github.com/dotnet/AspNetCore.Docs/blob/master/aspnetcore/host-and-deploy/directory-structure.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-hostdeploy**
* GitHub Login: @Rick-Anderson
* Microsoft Alias: **riande**

### PR DESCRIPTION
**Pull request by ericmutta**
Fixes #21344

For self-contained deployments (at least on ASP.NET Core 5.0) the `*.exe` extension is only generated for Windows. For Linux (and presumably macOS) no extension is generated. This behaviour is consistent with that for framework-dependent deployments so I just copied the relevant text. I am not sure if the behaviour applies to versions <= aspnetcore-3.0 so I haven't touched the text there.

It may also be useful to add that if you choose the `Produce single file` option when publishing, none of the `*.dll` files remain in the _publish_ directory (I love this feature!).

### CODE REVIEW

**Comment by Rick-Anderson** [comment]
Fixes #21344

**Comment by Rick-Anderson** [comment]
@jkotalik can you suggest a reviewer?

File: aspnetcore/host-and-deploy/directory-structure.md [hunk 1]
```diff
@@ -24,7 +24,7 @@ The *publish* directory contains the app's deployable assets produced by the [do
...
+| [Self-contained Deployment (SCD)](/dotnet/core/deploying/#self-contained-deployments-scd) | <ul><li>publish&dagger;<ul><li>Views&dagger; MVC apps, if views aren't precompiled</li><li>Pages&dagger; MVC or Razor Pages apps, if pages aren't precompiled</li><li>wwwroot&dagger;</li><li>\*.dll files</li><li>{ASSEMBLY NAME}.deps.json</li><li>{ASSEMBLY NAME}.dll</li><li>{ASSEMBLY NAME}{.EXTENSION} .exe extension on Windows, no extension on macOS or Linux</li><li>{ASSEMBLY NAME}.pdb</li><li>{ASSEMBLY NAME}.Views.dll</li><li>{ASSEMBLY NAME}.Views.pdb</li><li>{ASSEMBLY NAME}.runtimeconfig.json</li><li>web.config (IIS deployments)</li></ul></li></ul> |
```
Comment (Rick-Anderson) [inline aspnetcore/host-and-deploy/directory-structure.md:27]:
```suggestion
| [Self-contained Deployment (SCD)](/dotnet/core/deploying/#self-contained-deployments-scd) | <ul><li>publish&dagger;<ul><li>Views&dagger; MVC apps, if views aren't precompiled</li><li>Pages&dagger; MVC or Razor Pages apps, if pages aren't precompiled</li><li>wwwroot&dagger;</li><li>\*.dll files</li><li>{ASSEMBLY NAME}.deps.json</li><li>{ASSEMBLY NAME}.dll</li><li>{ASSEMBLY NAME}{.EXTENSION}.exe extension on Windows, no extension on macOS or Linux</li><li>{ASSEMBLY NAME}.pdb</li><li>{ASSEMBLY NAME}.Views.dll</li><li>{ASSEMBLY NAME}.Views.pdb</li><li>{ASSEMBLY NAME}.runtimeconfig.json</li><li>web.config (IIS deployments)</li></ul></li></ul> |
```

Comment (Rick-Anderson) [reply aspnetcore/host-and-deploy/directory-structure.md:27]: @ericmutta  extra space, right?

Comment (ericmutta) [reply aspnetcore/host-and-deploy/directory-structure.md:27]: Hi @Rick-Anderson I copied and pasted the text "as is", I believe the extra space was there in the [original](https://docs.microsoft.com/en-us/aspnet/core/host-and-deploy/directory-structure?view=aspnetcore-5.0):

image

File: aspnetcore/host-and-deploy/directory-structure.md [hunk 2]
```diff
@@ -24,7 +24,7 @@ The *publish* directory contains the app's deployable assets produced by the [do
 | App Type | Directory Structure |
 | -------- | ------------------- |
 | [Framework-dependent Executable (FDE)](/dotnet/core/deploying/#framework-dependent-executables-fde) | <ul><li>publish&dagger;<ul><li>Views&dagger; MVC apps; if views aren't precompiled</li><li>Pages&dagger; MVC or Razor Pages apps, if pages aren't precompiled</li><li>wwwroot&dagger;</li><li>\*.dll files</li><li>{ASSEMBLY NAME}.deps.json</li><li>{ASSEMBLY NAME}.dll</li><li>{ASSEMBLY NAME}{.EXTENSION} .exe extension on Windows, no extension on macOS or Linux</li><li>{ASSEMBLY NAME}.pdb</li><li>{ASSEMBLY NAME}.Views.dll</li><li>{ASSEMBLY NAME}.Views.pdb</li><li>{ASSEMBLY NAME}.runtimeconfig.json</li><li>web.config (IIS deployments)</li><li>createdump ([Linux createdump utility](https://github.com/dotnet/coreclr/blob/master/Documentation/botr/xplat-minidump-generation.md#configurationpolicy))</li><li>\*.so (Linux shared object library)</li><li>\*.a (macOS archive)</li><li>\*.dylib (macOS dynamic library)</li></ul></li></ul> |
```
Comment (Rick-Anderson) [inline aspnetcore/host-and-deploy/directory-structure.md:26]:
```suggestion
| [Framework-dependent Executable (FDE)](/dotnet/core/deploying/#framework-dependent-executables-fde) | <ul><li>publish&dagger;<ul><li>Views&dagger; MVC apps; if views aren't precompiled</li><li>Pages&dagger; MVC or Razor Pages apps, if pages aren't precompiled</li><li>wwwroot&dagger;</li><li>\*.dll files</li><li>{ASSEMBLY NAME}.deps.json</li><li>{ASSEMBLY NAME}.dll</li><li>{ASSEMBLY NAME}{.EXTENSION}.exe extension on Windows, no extension on macOS or Linux</li><li>{ASSEMBLY NAME}.pdb</li><li>{ASSEMBLY NAME}.Views.dll</li><li>{ASSEMBLY NAME}.Views.pdb</li><li>{ASSEMBLY NAME}.runtimeconfig.json</li><li>web.config (IIS deployments)</li><li>createdump ([Linux createdump utility](https://github.com/dotnet/coreclr/blob/master/Documentation/botr/xplat-minidump-generation.md#configurationpolicy))</li><li>\*.so (Linux shared object library)</li><li>\*.a (macOS archive)</li><li>\*.dylib (macOS dynamic library)</li></ul></li></ul> |
```

**Comment by Rick-Anderson** [comment]
@shirhatti @jkotalik @AngelosP 30 second review. This looks right to me.

**Pull request merged**